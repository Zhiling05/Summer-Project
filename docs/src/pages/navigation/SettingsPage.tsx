// src/pages/navigation/SettingsPage.tsx
export default function SettingsPage() {
  return <h1>SettingsPage</h1>;
}
