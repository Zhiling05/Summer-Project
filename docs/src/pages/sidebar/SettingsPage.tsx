import React, { useState } from "react";
import Header from '../../components/Header';    // 使用 Header 组件
import { useFontSize } from './FontSizeContext'; // 引用正确路径

export default function SettingsPage() {
  const { fontSize, setFontSize } = useFontSize(); // 获取当前字体大小和更新方法

  const handleFontSizeChange = (size: "13px" | "18px" | "24px") => {
    setFontSize(size);
  };

  return (
    <div>
      <Header title="Settings Page" showBack={true} /> {/* 使用 Header 中的返回按钮 */}
      <h1>Settings Page</h1>

      <section>
        <h2>Font Size</h2>
        <p>Choose your preferred font size:</p>
        <button onClick={() => handleFontSizeChange("13px")}>Small</button>
        <button onClick={() => handleFontSizeChange("18px")}>Medium</button>
        <button onClick={() => handleFontSizeChange("24px")}>Large</button>
      </section>
    </div>
  );
}
