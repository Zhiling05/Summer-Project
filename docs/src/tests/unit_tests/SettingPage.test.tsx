import { render, screen, fireEvent } from '@testing-library/react';
import SettingsPage from '../../pages/sidebar/SettingsPage';

// Mock Header
jest.mock('../../components/Header', () => ({ title }: { title: string }) => (
  <div data-testid="header">{title}</div>
));

// Mock FontSizeContext
const mockSetFontSize = jest.fn();
jest.mock('../../pages/sidebar/FontSizeContext', () => ({
  useFontSize: () => ({
    fontSize: '18px',
    setFontSize: mockSetFontSize,
  }),
}));

describe('SettingsPage Component', () => {
  test('renders header and font size buttons', () => {
    render(<SettingsPage />);

    expect(screen.getByTestId('header')).toHaveTextContent('Settings Page');
    expect(screen.getByRole('heading', { name: 'Settings Page' })).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: 'Font Size' })).toBeInTheDocument();
    expect(screen.getByText('Choose your preferred font size:')).toBeInTheDocument();

    expect(screen.getByRole('button', { name: 'Small' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Medium' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Large' })).toBeInTheDocument();
  });

  test('clicking font size buttons calls setFontSize with correct values', () => {
    render(<SettingsPage />);

    fireEvent.click(screen.getByText('Small'));
    expect(mockSetFontSize).toHaveBeenCalledWith('13px');

    fireEvent.click(screen.getByText('Medium'));
    expect(mockSetFontSize).toHaveBeenCalledWith('18px');

    fireEvent.click(screen.getByText('Large'));
    expect(mockSetFontSize).toHaveBeenCalledWith('24px');
  });
});
