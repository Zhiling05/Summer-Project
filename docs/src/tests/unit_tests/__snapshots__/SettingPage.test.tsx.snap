// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPage matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    SettingsPage
  </h1>
</DocumentFragment>
`;
